controller/model/identity_manager.go

```diff
@@ -145,6 +146,74 @@ func (self *IdentityManager) ApplyCreateWithEnrollments(cmd *CreateIdentityWithE
 	})
 }
 
+func (self *IdentityManager) CreateWithAuthenticators(identity *Identity, authenticators []*Authenticator, ctx *change.Context) (string, []string, error) {
+	if identity.Id == "" {
```

Comment:
identity is a pointer. should a nil check come first or let it panic?

Reply:
Panic, this is internal.